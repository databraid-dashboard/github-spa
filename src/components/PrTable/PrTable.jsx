/* eslint-disable import/no-named-as-default */
import React, { Component } from 'react';
import { Table } from 'semantic-ui-react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import PrTableRow from '../PrTableRow/PrTableRow';
import { retrievePrs } from '../../actions/prActions';
import './PrTable.css';


function prComponents(prIds) {
  return prIds.sort((a, b) => a - b).map(id => <PrTableRow key={id} prId={id} />);
}

export class PrTable extends Component {
  componentDidMount() {
    this.props.retrievePrs();
  }

  render() {
    if (this.props.loadingPrTable) {
      return (
        <div>Loading Pull Requests</div>
      );
    }
    
    return (
      <Table celled fixed>
        <Table.Header>
          <Table.Row>
            <Table.HeaderCell colSpan="4" textAlign="center">Pull Requests</Table.HeaderCell>
          </Table.Row>

          <Table.Row>
            <Table.HeaderCell>Title</Table.HeaderCell>
            <Table.HeaderCell>Created</Table.HeaderCell>
            <Table.HeaderCell>Submitted By</Table.HeaderCell>
            <Table.HeaderCell>Meargeability</Table.HeaderCell>
          </Table.Row>
        </Table.Header>
        <Table.Body>
          {prComponents(this.props.prIds)}

        </Table.Body>
      </Table>
    );
  }
}

PrTable.propTypes = {
  retrievePrs: PropTypes.func.isRequired,
  prIds: PropTypes.arrayOf.isRequired,
  loadingPrTable: PropTypes.bool.isRequired,
};

export const mapStateToProps = state => ({
  prIds: state.pullRequests.ids,
  // prsById: state.pullRequests.prsById
  loadingPrTable: state.loadingPrTable,
});

export const mapDispatchToProps = dispatch => bindActionCreators({
  retrievePrs,
}, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(PrTable);
