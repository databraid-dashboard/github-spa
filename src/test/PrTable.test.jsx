
import React from 'react';

import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { PrTable, mapStateToProps, mapDispatchToProps } from '../components/PrTable/PrTable';

const state = {
  pullRequests: {
    ids: [246869449],
    prsById: {
      246869449: {
        submittedBy: [],
        created: '2017-07-31T20:17:36Z',
        title: 'disregard #2',

      },
    },
  },
  loadingPrTable: false,
};

xdescribe('PrTable component', () => {
  it('should render a component with props as specified ', () => {
    const retrievePrs = jest.fn();
    const component = shallow(
      <PrTable prIds={state.pullRequests.ids} loadingPrTable={false} retrievePrs={retrievePrs} />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it('map\'s given State To Props ', () => {
    const expected = {
      prIds: [246869449],
      loadingPrTable: false,
    };
    expect(mapStateToProps(state)).toEqual(expected);
  });

  it('maps component dispatches to props', () => {
    const dispatch = jest.fn();
    expect(mapDispatchToProps(dispatch)).toHaveProperty('retrievePrs');
  });
  // it('Should have a Title: Pull Request', () => {
  //
  //   const component = shallow(
  //     <PrTable prIds={state.pullRequests.ids} loadingPrTable={false}/>,
  //   );
  //   console.log(component.find('Table.HeaderCell'));
  //   expect(component.find('.Table.HeaderCell').exists()).toBe(true);
  // })
});

// .ShallowWrapper.root.ShallowWrapper.unrendered
