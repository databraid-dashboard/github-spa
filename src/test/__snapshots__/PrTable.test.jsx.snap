// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrTable component should render a component with props as specified  1`] = `
<Table
  as="table"
  celled={true}
  fixed={true}
>
  <TableHeader
    as="thead"
  >
    <TableRow
      as="tr"
      cellAs="td"
    >
      <TableHeaderCell
        as="th"
        colSpan="4"
        textAlign="center"
      >
        Pull Requests
      </TableHeaderCell>
    </TableRow>
    <TableRow
      as="tr"
      cellAs="td"
    >
      <TableHeaderCell
        as="th"
      >
        Title
      </TableHeaderCell>
      <TableHeaderCell
        as="th"
      >
        Created
      </TableHeaderCell>
      <TableHeaderCell
        as="th"
      >
        Submitted By
      </TableHeaderCell>
      <TableHeaderCell
        as="th"
      >
        Meargeability
      </TableHeaderCell>
    </TableRow>
  </TableHeader>
  <TableBody
    as="tbody"
  >
    <Connect(Component)
      prId={246869449}
    />
  </TableBody>
</Table>
`;
